controller/apps.go

```diff
@@ -0,0 +1,413 @@
+	}
+	return osioSpace.Attributes.Name, nil
+}
+
+func (c *AppsController) getNamespaceName(ctx context.Context) (*string, error) {
+	osioclient, err := c.getAndCheckOsioClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeSpaceAttr, err := osioclient.GetNamespaceByType(nil, "user")
+	if err != nil {
+		return nil, err
+	}
+	if kubeSpaceAttr == nil || kubeSpaceAttr.Name == nil {
+		return nil, errors.NewNotFoundError("namespace", "user")
+	}
+
+	return kubeSpaceAttr.Name, nil
+}
+
+func (c *AppsController) getKubeClient(ctx context.Context) (*KubeClient, error) {
+
+	// create Auth API login object
+	authClient, err := NewAuthClient(ctx, c.AuthURL)
+	if err != nil {
+		goa.LogInfo(ctx, "errror creating auth client:"+tostring(err))
+		return nil, err
+	}
+
+	// get the user definition (for cluster URL)
+	authUser, err := authClient.getAuthUser()
+	if err != nil {
+		goa.LogInfo(ctx, "error accessing Auth server"+tostring(err))
+		return nil, err
+	}
+
+	if authUser == nil || authUser.Data.Attributes.Cluster == nil {
+		goa.LogInfo(ctx, "error getting user from Auth server:"+tostring(authUser))
+		return nil, nil
+	}
+
+	// get the login token for the cluster OpenShift API
+	osauth, err := authClient.getAuthToken(*authUser.Data.Attributes.Cluster)
+	if err != nil {
+		goa.LogInfo(ctx, "error getting openshift credentials:"+tostring(err))
+		return nil, err
+	}
+
+	kubeURL := *authUser.Data.Attributes.Cluster
+	kubeToken := *osauth.AccessToken
+
+	kubeNamespaceName, err := c.getNamespaceName(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the cluster login object
+	kc, err := NewKubeClient(kubeURL, kubeToken, *kubeNamespaceName)
+	if err != nil {
+		return nil, err
+	}
+	return kc, nil
+}
+
+func (c *AppsController) getAndCheckKubeClient(ctx context.Context) (*KubeClient, error) {
+
+	kc, err := c.getKubeClient(ctx)
+	if err != nil {
+		goa.LogInfo(ctx, "didn't actually get a token")
+		return nil, errors.NewUnauthorizedError("openshift token")
+	}
+	return kc, nil
+}
+
+// SetDeployment runs the setDeployment action.
+func (c *AppsController) SetDeployment(ctx *app.SetDeploymentAppsContext) error {
+	// AppsController_SetDeployment: start_implement
+
+	if ctx.PodCount == nil {
+		// TODO this should be error 400 (bad request) not 404 (not found)
+		return errors.NewNotFoundError("parameter", "podCount")
+	}
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeSpaceName, err := c.getSpaceNameFromSpaceID(ctx, ctx.SpaceID)
+	if err != nil {
+		return err
+	}
+
+	oldCount, err := kc.ScaleDeployment(*kubeSpaceName, ctx.AppName, ctx.DeployName, *ctx.PodCount)
+	if err != nil {
+		return err
+	}
+
+	goa.LogInfo(ctx, "podcount was ", oldCount, " will be set to "+strconv.Itoa(*ctx.PodCount))
+	// AppsController_SetDeployment: end_implement
+	return ctx.OK([]byte{})
+}
+
+func genData(start int, end int, limit int, low int, high int) []*app.TimedIntTuple {
```

Comment:
could you put a comment to describe the purpose of this function ?

Reply:
ah! is it just hard-coded data for now ?

Reply:
Yes, this is the last remaing stub while ebaron works on the implementation.  
I have added a comment.